File: e201.jl

function bin(n)
    for i=0:50
        mask = 1<<i
        print( (n&mask == mask) ? 1 : 0 )
    end
end

magic = 0
for i=51:100
    magic+=i*i
end


curhs = IntSet(0)
arr = zeros(Int64, magic+1)
arr[1] = 1
check = (1<<50)
guard = (1<<51)-1

for i=1:100
    ii = i*i
    newhs = IntSet()
    newarr = deepcopy(arr)
    for n in curhs
        cur = n+ii
        val = arr[n+1]
        if (cur > magic || val == 0)
            continue
        end

        if(cur == magic)
            bin(val)
            print("->",n,":",i)
            println()
        end

        newarr[cur+1] = newarr[cur+1] | ((val<<1)&guard)
        push!(newhs, n)
        push!(newhs, cur)
    end
    curhs = newhs
    arr = deepcopy(newarr)
end
counter = 0

#for i=51:100
#    counter+=i*i
#    println(counter,":",arr[counter+1])
#end

sum = 0

for i= 1:magic
    if (arr[i+1]&check) == check        
        sum+=(i)
    end
end

println(length(curhs))
#for n = magic-101:magic
#    bin(arr[n+1])
#    println("->",n)
#end

println("soln")
bin(arr[magic])
println()
bin(arr[magic+1])
println()
println(sum)

println(115039000)

File: e201.java
import java.util.*;

public class e201
{
    public static void main (String [] args)
    {
        int MAGIC = 295425;
        long check = 1L<<50;
        long [] arr = new long [295425+1];

        HashSet<Integer> hs = new HashSet<Integer>();
        hs.add(0);
        arr[0] = 1L;

        for(int i=1; i<=3; i++)
        {
            int ii = i*i;
            HashSet<Integer> new_hs = new HashSet<Integer>();
            for(Integer n:hs)
            {
                int cur = n+ii;
                if(cur == MAGIC)
                {
                    System.out.print(n+":"+i+":");
                    System.out.println(arr[cur]);
                }

                long val = arr[n];
                if(cur>=arr.length || ((val*2L)>(check))) continue;
                arr[cur] = (arr[cur] | (val*2L));
                new_hs.add(n);
                new_hs.add(cur);
            }
            hs = new_hs;
            for(Integer n: hs)
                System.out.println(n+":"+arr[n]);
            System.out.println();
        }
/*
        long sum = 0L;
        for(int i=1; i<arr.length; i++)
            if((arr[i] & check )== check)
            {
                sum+=i;
            }

        System.out.println(sum);
        System.out.println(check);
        System.out.println((arr[MAGIC] & check));
*/
    }


}
